### ISSUE TITLE
Importing library in Next.js fails

### ISSUE THREAD

**Issue by pooleja** [onfido/onfido-sdk-ui#733]
I am using Next.js and trying to load the component library.  The code below trigger the `require` statement when a button is clicked (to verify that SSR is not at play).  

This is the latest version of the sdk (5.2.2).  Older 3.x versions worked fine with this approach.

Any ideas why it is failing?

```
buttonClickHandler = () => {
...
const Onfido = require('onfido-sdk-ui')
...
}
```

```
Uncaught TypeError: Cannot read property 'bind' of undefined
    at eval ([absolute-resource-path]:1)
    at Object.eval ([absolute-resource-path]:1)
    at eval (index.js:44)
    at Object../node_modules/onfido-sdk-ui/lib/index.js (user-status.js:4504)
    at __webpack_require__ (manifest.js:715)
    at fn (manifest.js:118)
    at ProxyComponent.UserStatus._this.openOnFidoModal ([absolute-resource-path]:137)
    at ProxyComponent.UserStatus._this.handleIdentityClick ([absolute-resource-path]:169)
    at onClick ([absolute-resource-path]:233)
    at HTMLUnknownElement.callCallback ([absolute-resource-path]:149)
```

**Comment by rfreitas** [comment]
what version of Next.js are you using?
Can you share a sample project that we can take a look at?

**Comment by pooleja** [comment]
This is using all the latest next.js infra.  We saw the same issue on v 7 on Next.js as well.

Here is a repo that shows the error.  You click the button and the import fails once you update to the latest sdk ui.
https://github.com/pooleja/onfido-example/tree/master

**Comment by stefaniacardenas** [comment]
Hi @pooleja,
After some investigation, I believe there is a bug in the way Webpack 4 exports the package. We will try to fix it on our side as soon as possible but meanwhile, I would suggest to use the following workaround:
```
import React from 'react'
const onfidoContainerId = 'onfido-mount'
let Onfido;

export class Index extends React.Component {
  componentDidMount() {
    Onfido = require('onfido-sdk-ui/dist/onfido.min.js')
  }

  async handleClick() {
    console.log('About to require')
    Onfido.init({
        useModal: false,
        token: 'your token',
        onComplete: function() {
          // callback for when everything is complete
          console.log("Everything is complete");
        },
        steps: [
          'welcome',
          'document',
          'face',
          'complete'
        ]
      })
  }

  render () {
    return (
      <div>
        <p>Hello Next.js</p>
        <div id={onfidoContainerId} />
        <button onClick={this.handleClick}>
          Try to Require
        </button>
      </div>
    )
  }
}

export default Index
```

You will have to import the css file separately from `onfido-sdk-ui/dist/style.css` and put it in the head of the document.
I hope this helps.

**Comment by jamesmosier** [comment]
Will this issue be resolved in 5.3.0? I saw the line item `Public: Unbundled dependencies for npm` which I wasn't sure covered this or not.

**Comment by stefaniacardenas** [comment]
Yes, it will. I will reference this issue and 2 more related issues in the changelog.

**Comment by stefaniacardenas** [comment]
Hi, this issue has been fixed in release [5.3.0](https://github.com/onfido/onfido-sdk-ui/releases/tag/5.3.0).

Please import the library asynchronously as in the following example. 
```
let Onfido;
async componentDidMount() {
      Onfido = await import('onfido-sdk-ui')
}
Onfido.init()
```
Other types of import will probably break because Next.js executes code server-side first where `window` is not available, as stated in the Next.js documentation (see below).

> Next.js is universal, which means it executes code first server-side, then client-side. The window object is only present client-side, so if you absolutely need to have access to it in some React component, you should put that code in componentDidMount. This lifecycle method will only be executed on the client. You may also want to check if there isn't some alternative universal library which may suit your needs.

I will close this issue for now. Please feel free to reopen if you find any other problem.
